fc/protocol.py

```diff
@@ -660,24 +670,3 @@ def get_path(self, base_path, path):
             new_path = os.path.join(library, path)
         return new_path
 
-    def set_indent_level(self, indent_level):
-        """Set the level of indentation to use for progress output."""
-        self.indent_level = indent_level
-
-    def log_progress(self, *args):
```

Comment:
In getting rid of these helper methods, you're no longer indenting the log output from nested protocols, which makes the output harder to follow. Could we recover that functionality, or something equivalent?

Reply:
I'm not sure!

Happy to drop this PR and issue if we prefer the old way.

I'm guessing you could write a better log handler in the `fc-runner` that splits the logged messages into files according to their source

Reply:
Problem is you've lost the info by the time it gets to the fc-runner.

I was anticipating keeping the `log_progress` etc methods but using the logging library inside them?

Reply:
Yeah I thought about that but then I'm not sure we're using the logging framework in a sensible way.

As far as I understand the idea is that log everything to some central point, with both a logging level and some identifier for the messages' origin. Whatever happens with the logged messages, including the formatting, is then the responsibility of a handler.

So if we're going to impose our own structure / formatting on top of that, I don't really see the point in using it in the first place